internal/provider/resource_consul_cluster.go

```diff
@@ -329,6 +352,13 @@ func resourceConsulClusterCreate(ctx context.Context, d *schema.ResourceData, me
 	// The peering happens within the secondary cluster create operation.
 	autoHvnToHvnPeering := d.Get("auto_hvn_to_hvn_peering").(bool)
 
+	// Convert ip_allowlist to consul model.
```

Comment:
For creates:
- Get `ip_allowlist`
-`buildIPAllowlist()` to validate there are no duplicates. This will also return an empty list if there are no ip_allowlist entries
- Send create request